src/main/java/com/icthh/xm/ms/configuration/repository/impl/JGitRepository.java

```diff
@@ -143,6 +143,10 @@ protected void cloneRepository() {
             CloneCommand cloneCommand = Git.cloneRepository().setURI(gitProperties.getUri())
                                                          .setDirectory(repositoryFolder);
             cloneCommand = setAuthorizationConfig(cloneCommand);
+            cloneCommand.setBranch(gitProperties.getBranchName());
+            if (gitProperties.getDepth() > 0) {
+                cloneCommand.setDepth(gitProperties.getDepth());
```

Comment:
any tests?

Reply:
done, add tests to check amount of commits after initialization of JGitRepo with -1 depth and 1